chanjo_report/server/extensions.py

```diff
@@ -1,8 +1,31 @@
-# -*- coding: utf-8 -*-
-"""Extensions module. Each extension is initialized in the app factory
-located in app.py
-"""
+from alchy import make_declarative_base, QueryModel, ManagerMixin
 from chanjo.store.models import BASE
-from flask_alchy import Alchy
+from flask_sqlalchemy import SQLAlchemy
+from greenlet import getcurrent as _get_ident
+
+
+class Alchy(SQLAlchemy, ManagerMixin):
+    """Flask extension that integrates alchy with Flask-SQLAlchemy."""
+
+    def __init__(
+        self, app=None, use_native_unicode=True, session_options=None, Model=None, metadata=None
+    ):
+        if session_options is None:
+            session_options = {}
+
+        session_options.setdefault("query_cls", QueryModel)
+        session_options.setdefault("scopefunc", _get_ident)
```

Comment:
🚀 
Well found, and a nifty solution!

For my learning, what else was it pointing to before? From someone who didn't even know one could change it in general, or that Alchy had a session scope in particular. 😊 And why Greenlet? The current is not exposed in the corresponding native Python library?

Perhaps just add some attribution to the original Alchy class as well. In an ideal world this would have been a PR to `flask_alchy`, but that doesn't seem possible atm.

Reply:
Now I did!